### ISSUE TITLE
Remove Dependency Android Resolver (Android Advertising ID AAID)

### ISSUE THREAD

**Issue by ArnauKokoro** [firebase/firebase-unity-sdk#219]
Hi,

As a 1st April 2022 Google's permissions changes. AAID is not supported on Families Policies for children apps. So I have to remove every dependency which could use it from my app.

I could add a meta tag on the Android's Manifest to disable Analytics for recollect AAID, but I think my main problem now is that firebase-unity library also installs this dependency: com.google.android.gms.play-services-ads-identifier which seems to be causing the caution message about using AdMob on Play Console and is related for my app to not being accepted by google reviewer because it uses AAID.

My question here is: Is there a way to remove a single dependency from Android Resolver dependencies generated by the firebase-unity libraries installation?

I tried to remove it from AndroidResolverDependencies.xml but every time I make a build it's recreated.

Thanks!

**Comment by jonsimantov** [comment]
Thanks, we'll investigate. For now, if you run `gradlew -q dependencies app:dependencies` on your Android project, you can see which libraries are pulling in that dependency.

Once you know which libraries are pulling it in, you may be able to modify the __build.gradle__ file to remove the dependency directly, via something like this:
```
implementation('com.google.firebase:firebase-analytics') {
    exclude module: "play-services-ads-identifier"
}
```
(You can have multiple `exclude module: "play-services-ads-identifier"` lines to exclude additional modules.)

Also, have you tried actually adding that manifest meta tag? That may already turn off AAID enough to satisfy Play Console's requirement.

**Comment by ArnauKokoro** [comment]
Thanks @jonsimantov.

I did add that meta tag with false flag to disable AAID on Analytics collects.

I could remove play-services-ads-identifier from the project. Now there is not that dependency on the gradle. But Google still not admit the app.

These are my gradle dependencies:

implementation(name: 'com.google.android.datatransport.transport-api-3.0.0', ext:'aar')
implementation(name: 'com.google.android.datatransport.transport-backend-cct-3.0.0', ext:'aar')
implementation(name: 'com.google.android.datatransport.transport-runtime-3.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-auth-16.0.1', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-auth-api-phone-17.4.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-auth-base-16.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-base-17.6.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-basement-17.6.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-cloud-messaging-17.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-measurement-20.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-measurement-api-20.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-measurement-base-20.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-measurement-impl-20.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-measurement-sdk-20.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-measurement-sdk-api-20.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-safetynet-17.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-stats-17.0.0', ext:'aar')
implementation(name: 'com.google.android.gms.play-services-tasks-17.2.1', ext:'aar')
implementation(name: 'com.google.android.play.core-1.10.3', ext:'aar')
implementation(name: 'com.google.firebase.firebase-abt-21.0.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-analytics-20.0.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-analytics-unity-8.8.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-app-unity-8.8.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-auth-21.0.1', ext:'aar')
implementation(name: 'com.google.firebase.firebase-auth-interop-20.0.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-auth-unity-8.8.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-common-20.0.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-components-17.0.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-config-21.0.1', ext:'aar')
implementation(name: 'com.google.firebase.firebase-config-unity-8.8.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-datatransport-18.0.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-encoders-json-18.0.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-iid-21.1.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-iid-interop-17.1.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-measurement-connector-19.0.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-messaging-23.0.0', ext:'aar')
implementation(name: 'com.google.firebase.firebase-messaging-unity-8.8.0', ext:'aar')


Do you know if some dependency could use AAID?

Many thanks !!

**Comment by jonsimantov** [comment]
I don't know offhand which of those might use AAID, but if you run `gradlew -q dependencies :dependencies` from your Android project directory, it will print out a giant tree of all of the dependencies.

You should be able to search the output for `play-services-ads-identifier` to figure out how it's getting included.

It is possible to instruct gradle to omit a transitive dependency regardless of what brings it in - I can't guarantee that this will work in your case, but you can try it. Add this to your __build.gradle__ to omit the `play-services-ads-identifier` dependency from all configurations:
```
configurations.implementation {
  exclude group: 'com.google.android.gms', module: 'play-services-ads-identifier'
}
```
Once you do this, you should no longer see that dependency in the list generated by the above gradle command.

**Comment by ArnauKokoro** [comment]
@jonsimantov I've got approved after removing `play-services-ads-identifier` as a dependency on `AndroidResolverDependencies.xml` and disabling `AutoResolveOnBuild` on on External Dependency Manager - Android Resolver Settings. I also added `<meta-data android:name="google_analytics_adid_collection_enabled" android:value="false" />` to `AndroidManifest.xml` but the problem seems to be on `proguard` where I had specifications pointing to `play-services-ads-identifier`. So Google Systems thought I was using that dependency.

Thanks!!
